layouts/default.vue

```diff
@@ -0,0 +1,95 @@
+<script>
+import { useProgressStore } from '../stores/ProgressStore';
+import { mapStores } from 'pinia';
+import { useThemeChanger } from '~/composables/themeChanger';
+
+export default {
+	mounted() {
+		if (!this.progressStore.accessGranted(this.progressStore.currentLevel)) {
+			this.progressStore.jumpToLevel(this.progressStore.progress + 1, this.$router);
+		}
+
+		if (!this.themeInitialized()) {
+			this.initialize();
+		}
+
+		if (!this.progressStore.initialized) {
+			this.progressStore.init();
+		}
+	},
+
+	computed: {
+		...mapStores(useProgressStore),
+	},
+	methods: {
+		...useThemeChanger(),
+		resetProgress() {
+			this.progressStore.clearProgress();
+			this.progressStore.jumpToLevel(1, this.$router);
+		},
+
+	},
+};
+</script>
+
+<template>
+	<div class="flex flex-col dark:bg-blue-1-dark bg-blue-1-light h-screen">
+		<header class="flex justify-between w-11/12 h-20 mt-6 mx-auto mb-6 items-center">
+			<div class="pr-9 flex items-center">
+				<NuxtLink to="/">
+					<picture class="h-11 dark:hidden">
+						<source srcset="~\assets\img\icons\light\accessolotllogo-light.svg">
+						<img src="assets\img\icons\light\accessolotllogo-light.png">
+					</picture>
+					<picture class="h-11 hidden dark:block">
+						<source srcset="assets\img\icons\dark\accessolotllogo-dark.svg">
+						<img src="assets\img\icons\dark\accessolotllogo-dark.png">
+					</picture>
+				</NuxtLink>
+			</div>
+			<div class="flex grow items-center dark:bg-blue-2-dark bg-blue-2-light h-16 rounded-2xl">
+				<button type="button" class=" ml-12 invisible">
+					<Icon name="material-symbols:wb-sunny-outline" class="text-4xl" />
+				</button>
+				<nav class="flex justify-center items-center align-center flex-grow gap-2">
+					<NuxtLink :to="progressStore.prevLevelString" @click="progressStore.decrementLevel">
+						<Icon name="material-symbols:arrow-back-ios-rounded"
+							class="dark:text-text-dark text-text-light text-2xl" />
+					</NuxtLink>
+					<span class="text-2xl dark:text-text-dark text-text-light">Level {{ progressStore.currentLevel
+						}} of {{
+						progressStore.levelCount
+					}}</span>
```

Comment:
Den ganzen Text in einer Zeile wär zum Codelesen einfacher, aber vielleicht formatiert das die Autoformatierung anders, idk
```suggestion
					<span class="text-2xl dark:text-text-dark text-text-light">
					        Level {{ progressStore.currentLevel }} of {{ progressStore.levelCount }}
					</span>
```